src/main/java/org/elasticsearch/script/expression/FieldDataValueSource.java

```diff
@@ -49,13 +48,24 @@ public FunctionValues getValues(Map context, LeafReaderContext leaf) throws IOEx
     }
 
     @Override
-    public boolean equals(Object other) {
-        return fieldData.equals(other);
+    public boolean equals(Object o) {
```

Comment:
But there is literally code here that is never executed (null path). Why not regenerate with the given invariants like for the others?